src/lib/Extension.ts

```diff
@@ -49,23 +51,57 @@ export default class Extension {
         targetDirectory: string,
         answers: Object,
     ) {
+        const tempDirectory = targetDirectory + '/temp';
         try {
             if (!fs.existsSync(targetDirectory)) {
                 createDirectory(targetDirectory);
+                createDirectory(tempDirectory);
             }
-            await execa('git', ['init'], { cwd: targetDirectory });
+            await execa('git', ['init'], { cwd: tempDirectory });
             await execa(
                 'git',
                 ['remote', 'add', 'origin', answers.project_url],
-                { cwd: targetDirectory },
+                { cwd: tempDirectory },
             );
-            await execa('git', ['pull', 'origin', 'main:main'], {
-                cwd: targetDirectory,
+            
+            await execa('git', ['pull', '--recurse-submodules', 'origin', 'main:main'], {
+                cwd: tempDirectory,
             });
-            rimraf.sync(`${targetDirectory}/.git`); // unmark as git repo
+            await execa('git', ['submodule', 'update', '--init', '--recursive'], {
+                cwd: tempDirectory,
+            });
+            Debug("Fetching submodule path")
+            const s = await execa('git', ['config', '--file', '.gitmodules', '--get-regexp', 'path'], {
+                cwd: tempDirectory,
+            }).catch((err) => {
+                Debug("No submodule found")
+                return err;
+            });
+            const submodulePath = s?.stdout?.split?.(" ")?.[1] ?? null;
+            
+            rimraf.sync(`${tempDirectory}/.git`); // unmark as git repo
+            rimraf.sync(`${tempDirectory}/.gitmodules`); // Remove the .gitmodules file
+            
+            const submoduleGitPath = `${tempDirectory}/${submodulePath}/.git`
+            submodulePath && Debug(`Deleting ${submoduleGitPath}`)
+            submodulePath && rimraf.sync(submoduleGitPath); // unmark as git repo from submodules
+            
+            // move project from temporary directory to main directory
+            const files = glob.sync(path.join(tempDirectory, '{*,.*}'));
```

Comment:
Create common helper function with doc string comments and details of parameters @manishprajapati52

Reply:
Done